ikarus/finiteelements/mechanics/kirchhoffloveshell.hh

```diff
@@ -279,6 +286,30 @@ protected:
     K.template triangularView<Eigen::StrictlyLower>() = K.transpose();
   }
 
+  template <typename ScalarType>
+  void calculateMassImpl(const Requirement& par, const MatrixAffordance& affordance,
+                         typename Traits::template MatrixType<> M) const {
+    const auto geo  = underlying().localView().element().geometry();
+    const auto rhoT = thickness_ * density_;
+
+    for (const auto& [gpIndex, gp] : localBasis_.viewOverIntegrationPoints()) {
+      const auto intElement = geo.integrationElement(gp.position()) * gp.weight();
+      auto& N               = localBasis_.evaluateFunction(gpIndex);
+
+      auto nopI = Eigen::Matrix<double, worldDim, worldDim>::Zero().eval();
+      auto nopJ = Eigen::Matrix<double, worldDim, worldDim>::Zero().eval();
+
+      for (size_t i = 0; i < numberOfNodes_; ++i) {
+        nopI.diagonal().setConstant(N[i]);
+        for (size_t j = 0; j < numberOfNodes_; ++j) {
+          nopJ.diagonal().setConstant(N[j]);
+          M.template block<worldDim, worldDim>(i * worldDim, j * worldDim) +=
```

Comment:
I think it might be faster to just write:

```cpp
using MatrixType = Eigen::Matrix<double, worldDim, worldDim>;
M.template block<worldDim, worldDim>(i * worldDim, j * worldDim) += (N[i]*N[j]
               rhoT * intElement)* MatrixType::Identity() ;
```

or even simpler:


```cpp
using MatrixType = Eigen::Matrix<double, worldDim, worldDim>;
M.template block<worldDim, worldDim>(i * worldDim, j * worldDim).diagonal().array() += N[i]*N[j]
               rhoT * intElement;
```


or the most expressive: 

```cpp
#include <unsupported/Eigen/KroneckerProduct>
    auto M = Eigen::kroneckerProduct((N * N.transpose()).eval(), Eigen::Matrix3d::Identity()*intElement*rho).eval();
```
to create the matrix in one-shot. This is the most performant one. I assume.
https://godbolt.org/z/bsT6asEfr

Reply:
Nice